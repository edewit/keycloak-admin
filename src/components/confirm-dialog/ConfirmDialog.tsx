import React, { ReactNode, useState } from "react";
import { Button, ButtonVariant, Modal } from "@patternfly/react-core";
import { useTranslation } from "react-i18next";

export type ConfirmDialogProps = {
  titleKey: string;
  messageKey?: string;
  cancelButtonLabel?: string;
  continueButtonLabel?: string;
  continueButtonVariant?: ButtonVariant;
  onConfirm: () => void;
  onCancel?: () => void;
  children?: ReactNode;
};

export const ConfirmDialog = ({
  titleKey,
  messageKey,
  cancelButtonLabel,
  continueButtonLabel,
  continueButtonVariant,
  onConfirm,
  onCancel,
  children,
}: ConfirmDialogProps) => {
  const { t } = useTranslation();
  const [open, setOpen] = useState(true);
  return (
    <Modal
      title={t(titleKey)}
      isOpen={open}
      onClose={() => setOpen(false)}
      actions={[
        <Button
          id="modal-confirm"
          key="confirm"
          variant={continueButtonVariant || ButtonVariant.danger}
          onClick={() => {
            onConfirm();
            setOpen(false);
          }}
        >
          {t(continueButtonLabel || "common:continue")}
        </Button>,
        <Button
          id="modal-cancel"
          key="cancel"
          variant={ButtonVariant.secondary}
          onClick={() => {
            if (onCancel) onCancel();
            setOpen(false);
          }}
        >
          {t(cancelButtonLabel || "common:cancel")}
        </Button>,
      ]}
    >
      {!messageKey && children}
      {messageKey && t(messageKey)}
    </Modal>
  );
};
