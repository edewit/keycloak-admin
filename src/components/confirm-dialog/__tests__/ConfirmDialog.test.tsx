import React from "react";
import { mount } from "enzyme";
import { ConfirmDialog } from "../ConfirmDialog";

describe("Confirmation dialog", () => {
  it("renders simple confirm dialog", () => {
    const onConfirm = jest.fn();
    const simple = mount(
      <ConfirmDialog
        titleKey="Delete app02?"
        messageKey="If you delete this client, all associated data will be removed."
        continueButtonLabel="Delete"
        onConfirm={onConfirm}
      />
    );

    const button = simple.find("#modal-confirm").find("button");
    expect(button).not.toBeNull();
    
    button!.simulate("click");
    expect(onConfirm).toBeCalled();

    expect(simple).toMatchSnapshot();
  });
});
