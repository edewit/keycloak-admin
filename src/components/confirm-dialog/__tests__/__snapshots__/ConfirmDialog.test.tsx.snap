// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation dialog renders simple confirm dialog 1`] = `
<ConfirmDialog
  continueButtonLabel="Delete"
  messageKey="If you delete this client, all associated data will be removed."
  onConfirm={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  titleKey="Delete app02?"
>
  <Modal
    actions={
      Array [
        <Button
          id="modal-confirm"
          onClick={[Function]}
          variant="danger"
        >
          Delete
        </Button>,
        <Button
          id="modal-cancel"
          onClick={[Function]}
          variant="secondary"
        >
          cancel
        </Button>,
      ]
    }
    appendTo={[Function]}
    aria-describedby=""
    aria-label=""
    aria-labelledby=""
    className=""
    hasNoBodyWrapper={false}
    isOpen={false}
    onClose={[Function]}
    ouiaSafe={true}
    showClose={true}
    title="Delete app02?"
    variant="default"
  >
    <Portal
      containerInfo={<div />}
    >
      <ModalContent
        actions={
          Array [
            <Button
              id="modal-confirm"
              onClick={[Function]}
              variant="danger"
            >
              Delete
            </Button>,
            <Button
              id="modal-cancel"
              onClick={[Function]}
              variant="secondary"
            >
              cancel
            </Button>,
          ]
        }
        aria-describedby=""
        aria-label=""
        aria-labelledby=""
        boxId="pf-modal-part-0"
        className=""
        descriptorId="pf-modal-part-2"
        hasNoBodyWrapper={false}
        isOpen={false}
        labelId="pf-modal-part-1"
        onClose={[Function]}
        ouiaId="OUIA-Generated-Modal-default-1"
        ouiaSafe={true}
        showClose={true}
        title="Delete app02?"
        variant="default"
      />
    </Portal>
  </Modal>
</ConfirmDialog>
`;

exports[`confimation dialog renders simple confirm dialog 1`] = `
<ConfirmDialog
  continueButtonLabel="Delete"
  messageKey="If you delete this client, all associated data will be removed."
  onConfirm={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  titleKey="Delete app02?"
>
  <Modal
    actions={
      Array [
        <Button
          id="modal-confirm"
          onClick={[Function]}
          variant="danger"
        >
          Delete
        </Button>,
        <Button
          id="modal-cancel"
          onClick={[Function]}
          variant="secondary"
        >
          cancel
        </Button>,
      ]
    }
    appendTo={[Function]}
    aria-describedby=""
    aria-label=""
    aria-labelledby=""
    className=""
    hasNoBodyWrapper={false}
    isOpen={false}
    onClose={[Function]}
    ouiaSafe={true}
    showClose={true}
    title="Delete app02?"
    variant="default"
  >
    <Portal
      containerInfo={<div />}
    >
      <ModalContent
        actions={
          Array [
            <Button
              id="modal-confirm"
              onClick={[Function]}
              variant="danger"
            >
              Delete
            </Button>,
            <Button
              id="modal-cancel"
              onClick={[Function]}
              variant="secondary"
            >
              cancel
            </Button>,
          ]
        }
        aria-describedby=""
        aria-label=""
        aria-labelledby=""
        boxId="pf-modal-part-0"
        className=""
        descriptorId="pf-modal-part-2"
        hasNoBodyWrapper={false}
        isOpen={false}
        labelId="pf-modal-part-1"
        onClose={[Function]}
        ouiaId="OUIA-Generated-Modal-default-1"
        ouiaSafe={true}
        showClose={true}
        title="Delete app02?"
        variant="default"
      />
    </Portal>
  </Modal>
</ConfirmDialog>
`;
